crates/revm/src/optimism/fast_lz.rs

```diff
@@ -0,0 +1,314 @@
+        assert_eq!(flz_compress_len(&contract_call_tx_bytes), 471);
+    }
+
+    #[test]
+    fn test_flz_compress_len_no_repeats() {
+        let mut input = Vec::new();
+        let mut len = 0;
+
+        for i in 0..256 {
+            input.push(i as u8);
+            let prev_len = len;
+            len = flz_compress_len(&input);
+            assert!(len > prev_len);
+        }
+    }
+
+    #[test]
+    fn test_flz_solady_parity() {
+        for _ in 0..100 {
+            let mut input = [0; 1000000];
+            ethers_core::rand::thread_rng().fill_bytes(&mut input);
+            assert_eq!(
+                flz_compress_len(&input),
+                solady_flz_compress(&input),
+                "{:?}",
+                input
+            );
+        }
+    }
+
+    #[test]
+    fn test_flz_native_evm_parity() {
+        // This bytecode and ABI is for a contract, which wraps the LibZip library for easier fuzz testing.
+        // The source of this contract is here: https://github.com/danyalprout/fastlz/blob/main/src/FastLz.sol#L6-L10
+        let abi = BaseContract::from(
+            parse_abi(&["function fastLz(bytes) external view returns (uint256)"]).unwrap(),
+        );
```

Comment:
This shoudl be using alloy cc @DaniPopes 

```
alloy_sol_types::sol! {
    interface FaztLz {
       function fastLz(bytes) external view returns (uint256);
   }
}
```

And then `fastLzCall::abi_encode`

https://docs.rs/alloy-sol-types/latest/alloy_sol_types/

Reply:
Thanks, I'll apply that refactor to the [examples](https://github.com/bluealloy/revm/blob/main/examples/fork_ref_transact.rs#L37-L45) as well if this is the preferred method for interacting with solidity types